tasks/DigitalOutMultiplexerTask.cpp

```diff
@@ -0,0 +1,83 @@
+/* Generated from orogen/lib/orogen/templates/tasks/Task.cpp */
+
+#include "DigitalOutMultiplexerTask.hpp"
+#include "base-logging/Logging.hpp"
+#include "base/Time.hpp"
+#include "linux_gpios/linux_gpiosTypes.hpp"
+
+using namespace linux_gpios;
+using namespace motors_roboteq_canopen;
+using namespace std;
+
+DigitalOutMultiplexerTask::DigitalOutMultiplexerTask(std::string const& name)
+    : DigitalOutMultiplexerTaskBase(name)
+{
+}
+
+DigitalOutMultiplexerTask::~DigitalOutMultiplexerTask()
+{
+}
+
+/// The following lines are template definitions for the various state machine
+// hooks defined by Orocos::RTT. See DigitalOutMultiplexerTask.hpp for more detailed
+// documentation about them.
+
+bool DigitalOutMultiplexerTask::configureHook()
+{
+    if (!DigitalOutMultiplexerTaskBase::configureHook())
+        return false;
+    return true;
+}
+
+bool DigitalOutMultiplexerTask::startHook()
+{
+    if (!DigitalOutMultiplexerTaskBase::startHook())
+        return false;
+    return true;
+}
+
+void DigitalOutMultiplexerTask::updateHook()
+{
+    DigitalOutMultiplexerTaskBase::updateHook();
+
+    GPIOState overall_command;
+    overall_command.states.resize(2);
```

Comment:
raw_io::Digital is unfortunately not initializing its `data` field ... so you are sending random values if either one or the other port has NoData.

I would argue it is a good thing to modify raw_io::Digital...